### PR TITLE
test(rome_js_analyze): every rule has its folder

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

This PR normalizes the folder structure of linter tests: every rule has a dedicated folder.

Moreover, I caught a copy-paste error: `correctness/noDuplicateObjectKeys.js.snap` has been moved to `suspicious/noDuplicateObjectKeys.js.snap`. However, the original file was not removed. This kind of error should be easier to catch with the new structure.

I also turn some files into JSONC. This is easier to catch a potential missing diagnostic since the code and the diagnostic are close.

I also updated the CONTRIBUTING file to encourage the use of folder. By the way, we could change the test script to require it...

## Documentation

CONTRIBUTING updated.

- [ ] The PR requires documentation
- [ ] I will create a new PR to update the documentation

### CODE REVIEW

**Comment by ematipico** [comment]
Assuming that orphan snapshots is just an issue of `insta` - you can delete any testing assertion and the snapshot won't be deleted, and not of the testing infrastructure, what's this change going to solve/make better for us?

**Comment by Conaclos** [comment]
> what's this change going to solve/make better for us?

In my view it makes new contributor entrance easier because there is a single way to start writing tests. Moreover, it makes easier to identify in a tree view where the tests are located since everything is in a folder.

> What's the change around JSON? I can't understand the argument around the diagnostics

Baically with a js file you get the following snpshot:

````md
# Input
```js
let x = 1; foo(x)

let x = 1; x++;

for (let i in [1,2,3]) { foo(i); }
```

# Diagnostics
```
invalid.jsonc:1:1 lint/style/useConst  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! This 'let' declares a variable which is never re-assigned.
  
  > 1 │ let x = 1; foo(x);
      │ ^^^
  
  i 'x' is never re-assigned.
  
  > 1 │ let x = 1; foo(x);
      │     ^

invalid.jsonc:3:6 lint/style/useConst  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! This 'let' declares a variable which is never re-assigned.
  
  > 3 │ for (let i in [1,2,3]) { foo(i); }
      │      ^^^
  
  i 'i' is never re-assigned.
  
  > 1 │ for (let i in [1,2,3]) { foo(i); }
      │          ^

```
````

While with jsonc, you get the following snapshot:

````md
# Input
```js
let x = 1; foo(x);
```

# Diagnostics
```
invalid.jsonc:1:1 lint/style/useConst  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! This 'let' declares a variable which is never re-assigned.
  
  > 1 │ let x = 1; foo(x);
      │ ^^^
  
  i 'x' is never re-assigned.
  
  > 1 │ let x = 1; foo(x);
      │     ^
```

# Input
```js
let x = 1; x++;
```

# Input
```js
for (let i in [1,2,3]) { foo(i); }
```

# Diagnostics
```
invalid.jsonc:1:6 lint/style/useConst  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! This 'let' declares a variable which is never re-assigned.
  
  > 1 │ for (let i in [1,2,3]) { foo(i); }
      │      ^^^
  
  i 'i' is never re-assigned.
  
  > 1 │ for (let i in [1,2,3]) { foo(i); }
      │          ^
```
````

It is easier to catch in the second form that an input does not trigger the rule, and thus the test is either in the wrong file (in `invalid.jsonc` instead of `valid.jsonc`) or the rule implementation has an issue. This is because the inputs and diagnostics are grouped in the snapshot.

**Comment by ematipico** [comment]
I think there's something wrong with the code blocks, both have diagnostics coming from a `jsonc` file.

I would argue that having a JS is better with IDE experience, because you can see if you write wrong code.
Anyway, this seems to be bikeshedding and I don't want to stop the PR from being merged!

**Comment by ematipico** [comment]
I think there's something wrong with the code blocks, both have diagnostics coming from a `jsonc` file.

I would argue that having a JS file is better with IDE experience, because you can see if you write wrong code.
Anyway, this seems to be bikeshedding and I don't want to stop the PR from being merged!

**Comment by Conaclos** [comment]
> I think there's something wrong with the code blocks, both have diagnostics coming from a jsonc file.

Fixed. They are "hand-made" examples.

File: crates/rome_analyze/CONTRIBUTING.md [hunk 1]
```diff
@@ -146,16 +146,9 @@ _**your test cases are placed inside the wrong group, you won't see any diagnost
 
 Since each new rule will start from `nursery`, that's where we start.
 If you used `just new-lintrule`, a folder that use the name of the rule should exist.
+Otherwise, create a folder called `myRuleName/`, and then create various files where you want to create different cases.
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:149]: I suggest  "one or more files" instead. It explains that even a single file with all cases is fine.

File: crates/rome_analyze/CONTRIBUTING.md [hunk 2]
```diff
@@ -146,16 +146,9 @@ _**your test cases are placed inside the wrong group, you won't see any diagnost
...
-Otherwise, you have two options:
-
-- Create a single file called like the rule name, e.g. `myRuleName.js`, `myRuleName.tsx`, etc.
-  The extension of the file matters based on which kind of file you need to test;
-
-- Create a folder called `myRuleName/`, and then create various files where you want to create different cases.
-  These options are useful if your rules target different super languages, or you want to split your cases among different files.
-
-If you chose to create a folder, a common pattern is to create files prefixed by `invalid` or `valid`.
+a common pattern is to create files prefixed by `invalid` or `valid`.
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:151]: "A common...". Capital A?

**Review by ematipico — COMMENTED** [review]
Don't we have to change the test runner to make sure that only folders are allowed?

**Review by ematipico — APPROVED** [review]

**Pull request merged**